### DISCUSSION TITLE
Brownie Compile Doesn't Work: SolcInstallationError

### DISCUSSION THREAD

**Discussion by nilo5** [smartcontractkit/full-blockchain-solidity-course-py#1833] [Q&A] [answered]
Error: SolcInstallationError: Downloaded binary would not execute, or returned unexpected output. If this issue persists, you can try to compile from source code using `solcx.compile_solc('0.8.18')`.

There are already a couple of posts talking about this error, while I had successfully install solc through brew and added it to the global env, it still shows the SolcInstallationError error.  One interesting hypothesis I found is that the command is trying to download 64amd, while latest mac(which I am using) is 128amd.  Wonder if there is a way to configure that into the command?

I am currently at 4:32:55 in the course.  Thanks for the help in advance.

**Comment by kailashchoudhary11** [comment]
`solcx.compile_solc('0.8.18')` Did you try to use this?

**Reply by nilo5** [reply]
It still didn't work.   However, I found it works when I directly deploy it to a local test chain, just not to compile the contract.

**Comment by cromewar** [comment, marked as answer]
Hello there @nilo5 this shouldn't be a problem as the Rosetta layer should automatically translate those instructions. Could you please share your code here?

**Reply by nilo5** [reply]
I copy paste the code from https://github.com/PatrickAlphaC/brownie_simple_storage/blob/main/contracts/SimpleStorage.sol

And I only have this .sol file.

**Reply by cromewar** [reply]
I meant the code you are using to deploy this, your python script. Please share it here so we can give it a check.

**Reply by nilo5** [reply]
Hmm, I don't have a deploy code.  I only have this .sol file, and was using "brownie compile" as following the tutorial.  Do we really need a deploy file to compile the contract?

**Reply by cromewar** [reply]
Well if you are getting the error on the compiler itself, it means something is messing up with your brownie configuration, try the following:

1. delete the build folder on your project.
2. delete the .brownie file on your home folder.
3. select the correct compiler on the vscode extension (left click, select compiler version remote).
